react/features/base/tracks/middleware.js

```diff
@@ -326,6 +332,29 @@ function _handleNoDataFromSourceErrors(store, action) {
     }
 }
 
+/**
+ * Creates a fake participant for screen share using the track's source name as the participant id.
+ *
+ * @param {Store} store - The redux store in which the specified action is dispatched.
+ * @param {Action} action - The redux action dispatched in the specified store.
+ * @private
+ * @returns {void}
+ */
+function createFakeScreenShareParticipant({ dispatch, getState }, { track }) {
+    const state = getState();
+    const particiipantId = track?.jitsiTrack?.getParticipantId?.();
+    const particiipant = getParticipantById(state, particiipantId);
+    const name = particiipant?.name ? `${particiipant.name}'s screen` : 'Fellow Jitster\'s Screen';
+
+    dispatch(participantJoined({
+        conference: state['features/base/conference'].conference,
+        id: track.jitsiTrack.getSourceName(), // endpointID-video-idx
```

Comment:
re `// endpointID-video-idx` it's actually `//endpointId-v0` for the first one, so kind of hard to describe this way...